controllers/storagecluster/storageclasses.go

```diff
@@ -528,3 +528,38 @@ func getTopologyConstrainedPools(initData *ocsv1.StorageCluster) string {
 	}
 	return string(topologyConstrainedPoolsStr)
 }
+
+func getTopologyConstrainedPoolsExternalMode(data map[string]string) string {
+	type topologySegment struct {
+		DomainLabel string `json:"domainLabel"`
+		DomainValue string `json:"value"`
+	}
+	// TopologyConstrainedPool stores the pool name and a list of its associated topology domain values.
+	type topologyConstrainedPool struct {
+		PoolName       string            `json:"poolName"`
+		DomainSegments []topologySegment `json:"domainSegments"`
+	}
+	var topologyConstrainedPools []topologyConstrainedPool
+
+	domainLabel := data["topologyFailureDomainLabel"]
+	domainValues := strings.Split(data["topologyFailureDomainValues"], ",")
+	poolNames := strings.Split(data["topologyPools"], ",")
+
```

Comment:
Can you pls check if we can construct the structure directly from the json? using marshal/unmarshal etc.